docs/rfcs/0003/src/fence_hpe_nnf/fence_hpe_nnf.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+#
+# Copyright 2021, 2022 Hewlett Packard Enterprise Development LP
+# Other additional copyright holders may be indicated within.
+#
+# The entirety of this work is licensed under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys, re, time
+import atexit
+from os.path import exists
+
+sys.path.append("@FENCEAGENTSLIBDIR@")
+
+# In some cases, we do not wish to allow power off
+allow_powerOff = False
+
+try:
+	from fencing import *
+	from fencing import fail, EC_STATUS
+except:
+	if exists("/usr/share/fence/fencing.py"):
+		sys.path.append("/usr/share/fence/")
+		try:
+			from fencing import *
+			from fencing import fail, EC_STATUS
+		except:
+			fence_import_error()
+	else:
+		fence_import_error()
+		
+
+def fence_import_error():
+	print("Error importing fencing libs, may need to update python path")
+	print("e.g. export PYTHONPATH=/usr/share/fence/")
+	exit(1)
+
+def get_power_status(conn, options):
```

Comment:
Is this just fence_hpe_hss_ssh, but with different set_power_status() and reboot_cycle() methods? Looks like the remainder of the code (imports, get_power_status(), main()) is the same.

Reply:
Yes, I copied it from that agent and then neutered it a bit.  Now that you bring this up, this agent does query the Redfish CLI for the current power status, so it may be a candidate for a more native redfish approach rather than using SSH.

Reply:
:+1: a TODO for the next person

Reply:
I did take a quick look at the HSS, you can interact with it easily with curl, however the port is not currently listening on the external IP.